novawallet/Common/Services/DelegatedAccounts/SyncService/DelegatedAccountChainSyncService.swift

```diff
@@ -0,0 +1,407 @@
+            }
+        }
+    }
+
+    func createWalletsWrapper(
+        for filter: ProxySyncChainWalletFilter?,
+        chain: ChainModel
+    ) -> CompoundOperationWrapper<[ManagedMetaAccountModel]> {
+        let metaAccountsOperation = metaAccountsRepository.fetchAllOperation(with: .init())
+
+        let filterOperation = ClosureOperation<[ManagedMetaAccountModel]> {
+            let allWallets = try metaAccountsOperation.extractNoCancellableResultData()
+
+            guard let filter = filter else {
+                return allWallets
+            }
+
+            return allWallets.filter { filter(chain, $0.info) }
+        }
+
+        filterOperation.addDependency(metaAccountsOperation)
+
+        return CompoundOperationWrapper(targetOperation: filterOperation, dependencies: [metaAccountsOperation])
+    }
+
+    func createDelegatedAccountsListWrapper(
+        connection: JSONRPCEngine,
+        runtimeProvider: RuntimeProviderProtocol,
+        metaAccountsWrapper: CompoundOperationWrapper<[ManagedMetaAccountModel]>,
+        blockHash: Data?
+    ) -> CompoundOperationWrapper<[AccountId: [DelegatedAccount]]> {
+        let proxyRepository = ChainProxyAccountsRepository(
+            requestFactory: requestFactory,
+            connection: connection,
+            runtimeProvider: runtimeProvider,
+            blockHash: blockHash
+        )
+        let multisigRepository = MultisigAccountsRepository(chain: chainModel)
+
+        let delegatedListWrapper = createDelegatedAccountsListWrapper(
+            proxyRepository: proxyRepository,
+            multisigRepository: multisigRepository,
+            metaAccountsWrapper: metaAccountsWrapper,
+            chainModel: chainModel
+        )
+
+        return delegatedListWrapper
+    }
+
+    func createDelegatedAccountsListWrapper(
+        proxyRepository: DelegatedAccountsRepositoryProtocol,
+        multisigRepository: DelegatedAccountsRepositoryProtocol,
+        metaAccountsWrapper: CompoundOperationWrapper<[ManagedMetaAccountModel]>,
+        chainModel: ChainModel
+    ) -> CompoundOperationWrapper<[AccountId: [DelegatedAccount]]> {
+        let accountsListWrapper: CompoundOperationWrapper<[AccountId: [DelegatedAccount]]>
+        accountsListWrapper = OperationCombiningService.compoundNonOptionalWrapper(
+            operationQueue: operationQueue
+        ) { [weak self] in
+            guard let self else { throw BaseOperationError.parentOperationCancelled }
+
+            let chainMetaAccounts = try metaAccountsWrapper.targetOperation.extractNoCancellableResultData()
+
+            let possibleProxiedList: [AccountId] = chainMetaAccounts.compactMap { wallet in
+                guard wallet.info.type != .proxied else {
+                    return nil
+                }
+
+                return wallet.info.fetch(for: chainModel.accountRequest())?.accountId
+            }
+            let possibleMultisigList: [AccountId] = chainMetaAccounts.compactMap { wallet in
```

Comment:
I think you don't need delegated account wallets in the initial set at all since you want to detect also the case when the delegate is cancelled